lib/rspec/core.rb

```diff
@@ -129,6 +129,20 @@ def self.current_example=(example)
     RSpec::Support.thread_local_data[:current_example] = example
   end
 
+  # Get the current scope rspec is currently executing in
+  # Returns nil until it starts executing an example_group
+  # TODO ... something something actually what it returns nil when i call it in an example group?
```

Comment:
The first execution could set this to `:suite` and similarly reset it to `:suite` before the "after suite" hooks are run.

Reply:
Went w/ setting it to :suite as soon as rspec loads, and then setting it to nil after the suite and all hooks are done.